### PR TITLE
[Docs] Japanese translation of docs/cli_development.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add a Japanese translation of docs/cli_development.md

I accidentally removed branch during pull request of this file.
It caused the deletion of pull request.

I taged on my local repository just before the deletion, so I make this pull request from it.
Sorry for the trouble.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
元の pr は、#9513 ですね。

File: docs/ja/cli_development.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+| cli.log.debug | `{fg_cyan}☐{style_reset_all}` |
+| cli.log.notset | `{style_reset_all}¯\\_(o_o)_/¯` |
+
+### 出力 (`cli.echo`)
+
+場合によっては単にログシステムの外部でテキストを出力する必要があります。これは、固定データを出力したり、ログに記録してはいけない何かを書きだす場合に適しています。ほとんどの場合、`cli.echo` よりも `cli.log.info()` を選ぶべきです。
+
+### テキストの色付け
+
+テキスト内に色トークンを含めることで、テキストの出力を色付けすることができます。情報を伝えるためではなく、強調するために色を使いますユーザは色を無効にできることを覚えておいてください。色を無効にした場合でもサブコマンドは引き続き使えるようにしてください。
```
Comment (mtei) [inline docs/ja/cli_development.md:89]:
```suggestion
テキスト内に色トークンを含めることで、テキストの出力を色付けすることができます。情報を伝えるためではなく、強調するために色を使います。ユーザは色を無効にできることを覚えておいてください。色を無効にした場合でもサブコマンドは引き続き使えるようにしてください。
```

File: docs/ja/cli_development.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+背景色を設定するのは、あなたがやっていることに不可欠ではない限り、通常は避けるべきです。ターミナルの色に関しては、ユーザは多くの好みを持つため、黒と白の両方の背景に対してうまく機能する色を選択する必要があることを覚えておいてください。
```
Comment (mtei) [inline docs/ja/cli_development.md:91]:
```suggestion
背景色を設定するのは、あなたがやっていることに不可欠ではない限り、通常は避けるべきです。ユーザは、ターミナルの色に関しては多くの好みを持つため、あなたは黒と白のどちらの背景に対してもうまく機能する色を選択する必要があることを覚えておいてください。
```

File: docs/ja/cli_development.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+| {style_bright} | テキストを明るくする |
+| {style_dim} | テキストを暗くする |
+| {style_normal} | テキストを通常にする (`{style_bright}` または `{style_dim}` のどちらでもない) |
+| {style_reset_all} | 全てのテキストの属性をデフォルトに再設定する(これは自動的に全ての文字列の最後に自動的に追加されます。) |
+| {bg_reset} | 背景色をユーザのデフォルトに再設定します。 |
+| {fg_reset} | 背景色をユーザのデフォルトに再設定します。 |
+
+# 引数と設定
+
+QMK は引数の解析と設定の詳細を処理します。新しい引数を追加すると、サブコマンドの名前と引数の長い名前に基づいて設定ツリーに自動的に組み込まれます。属性形式のアクセス (`cli.config.<subcommand>.<argument>`) あるいは辞書形式のアクセス (`cli.config['<subcommand>']['<argument>']`) を使って、`cli.config` 内のこの設定にアクセスすることができます。
```
Comment (mtei) [inline docs/ja/cli_development.md:119]: 原文の 'for you' はこんなニュアンスなのではないかと想像します。

```suggestion
QMK は引数の解析と設定の詳細をあなたの代わりに処理します。新しい引数を追加すると、サブコマンドの名前と引数の長い名前に基づいて設定ツリーに自動的に組み込まれます。属性形式のアクセス (`cli.config.<subcommand>.<argument>`) あるいは辞書形式のアクセス (`cli.config['<subcommand>']['<argument>']`) を使って、`cli.config` 内のこの設定にアクセスすることができます。
```

File: docs/ja/cli_development.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+## 設定ファイルの書き方
+
+設定は変更しても書き出されません。ほとんどのコマンドでこれをする必要はありません。ユーザに `qmk config` を使って設定を慎重に変更させることをお勧めします。
+
+設定を書き出すために `cli.save_config()` を使うことができます。
+
+## 設定からの引数の除外
+
+一部の引数は設定ファイルに伝播すべきではありません。これらは引数を作成する時に `arg_only=True` を追加することで除外することができます。
```
Comment (mtei) [inline docs/ja/cli_development.md:175]: 「伝播」でも意味はわかるのですが、どうも普段こういう言い方はしない気がします。
```suggestion
一部の引数は設定ファイルに反映すべきではありません。これらは引数を作成する時に `arg_only=True` を追加することで除外することができます。
```

File: docs/ja/cli_development.md (hunk 3 above)
Comment (umi-umi) [reply docs/ja/cli_development.md]: 単純に「あなたのために」と訳してから、なんだかくどいなぁと削除してしまったのですが、
ご指摘いただいたとおりここで言いたいのは「代理」ですね。。ありがとうございます！

**Comment by mtei** [comment]
I think this is ready to merge.

**Comment by mtei** [comment]
Thanks !

**Pull request merged**